src/main.ts

```diff
@@ -30,18 +30,40 @@ async function main() {
         let servicePrincipalKey = secrets.getSecret("$.clientSecret", true);
         let tenantId = secrets.getSecret("$.tenantId", false);
         let subscriptionId = secrets.getSecret("$.subscriptionId", false);
+        let resourceManagerEndpointUrl = secrets.getSecret("$.resourceManagerEndpointUrl", false);
+        let profileVersion = core.getInput("profileVersion");
+        let customEnvironmentName = core.getInput("customEnvironmentName");
         const enableAzPSSession = core.getInput('enable-AzPSSession').toLowerCase() === "true";
         if (!servicePrincipalId || !servicePrincipalKey || !tenantId || !subscriptionId) {
             throw new Error("Not all values are present in the creds object. Ensure clientId, clientSecret, tenantId and subscriptionId are supplied.");
         }
         // Attempting Az cli login
+        if (customEnvironmentName != "") {

```

Comment:
Can we fix the customEnvironmentName to be AzureStack and check for that condition here?

Reply:
@rakku-ms - Could you please confirm if there is a strong requirement for knowing the Environment "name"? Or could we just seek Environment "type" which can just be an allowed list of keywords like "AzureStack" ?

Reply:
yes, A customer can have several instances of AzureStack and the name cannot be limited to a particular set.  You can see that when we do 'az cloud register' , the name is a parameter for customer to name it whatever they want.

Reply:
@bganapa  
- In our Azure DevOps world, we use a list of pre-defined values for the different cloud types. Users targeting Azure Stack, only needs to select AzureStack as the environment and provide the Azure Stack server URL.  

![89559574-7836a400-d833-11ea-9d64-8bcdafcdbfa1](https://user-images.githubusercontent.com/20986928/89562434-94d4db00-d837-11ea-8929-cce7950d06cb.png)

- The `az cloud register` persists for only one session and cannot be reused in a new session. Also, a user cannot login into multiple environments in a single workflow job. I am not sure if we are gaining anything by letting a user name the environment whatever they want. If the user wants to differentiate between the instances, he can specify it in the job name, something like :
```
name: Run on Azure Stack
jobs: 
  run-on-instance-1:
   - uses: azure/login@v1
        with:
          environment: AzureStack
          creds: ${{ secrets.AZURE_CREDENTIALS_FOR_INSTANCE_1 }}

   run-on-instance-2:
   - uses: azure/login@v1
        with:
          environment: AzureStack
          creds: ${{ secrets.AZURE_CREDENTIALS_FOR_INSTANCE_2 }}
```
- We also do not ask for profile version in Azure DevOps, just the Server Url. Can this value be fixed in the action instead of taking input from user.
 
![Capture](https://user-images.githubusercontent.com/20986928/89562913-4d9b1a00-d838-11ea-974d-0377ce6af66b.PNG)

Reply:
@asranja I am using self-hosted runner as Azure Stack env is not publicly accessible from internet. For me registered cloud is getting persisted on my desktop. So, I am unable to run workflow on 2 different Azure Stack environment with same environmentName `AzureStack`. The second run also happens against the first since the cloud with the name `AzureStack` already exists. The register cmd doesn't overwrite the previous environment and just try to login to the existing one. One workaround would be to first remove the already existing `AzureStack` environment and then try to register. If this is what your team prefers, I will make this change. FYI: @bganapa  
```
PS login> az cloud list --output table
IsActive    Name                  Profile
----------  --------------------  -----------------
False       AzureCloud            latest
False       AzureChinaCloud       latest
False       AzureUSGovernment     latest
False       AzureGermanCloud      latest
False       AzureStackRedmond001  2019-03-01-hybrid
False       AzureStackTest01      2019-03-01-hybrid
False       AzureStackTest02      2019-03-01-hybrid
False       AzureStackLH01        2019-03-01-hybrid
False       AzureStackRedmond     2019-03-01-hybrid
```

Reply:
@rakku-ms Thanks for giving more details regarding the environment name. We will discuss this internally and get back.

Regarding the profile version: As per my understanding, most Azure resources will have different api-version depending on the profile version. Currently, we have hard-coded all the api-versions in our REST Apis, for [example](https://github.com/Azure/sql-action/blob/71cbe30f683e929e6b48173468b877fb22e66009/src/AzureSqlResourceManager.ts#L114). I am not sure how are we planning on updating the rest api-versions depending on the profile version.

Reply:
@rakku-ms - Can you please help understand how many customers usually need to access two or more different Azure stack environments?